lib/chef/provider/windows_path.rb

```diff
@@ -0,0 +1,63 @@
+#
+# Author:: Nimisha Sharad (<nimisha.sharad@msystechnologies.com>)
+# Copyright:: Copyright 2008-2017, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "Win32API" if Chef::Platform.windows?
+require "chef/exceptions"
+
+class Chef
+  class Provider
+    class WindowsPath < Chef::Provider
+      ExpandEnvironmentStrings = Win32API.new("kernel32", "ExpandEnvironmentStrings", %w{ P P L }, "L") if Chef::Platform.windows? && !defined?(ExpandEnvironmentStrings)
```

Comment:
this is a new pattern.

https://github.com/cosmo0920/win32-api/blob/master/README#L55

>  The current Win32API library that ships with the standard library has been slated for removal from Ruby 2.0 and it will not receive any updates in the Ruby 1.8.x branch.

This ruby bug says a similar thing but isn't current: https://bugs.ruby-lang.org/issues/8601

https://github.com/cosmo0920/win32-api/blob/master/README#L109

> I really don't have future plans for this library since you should use FFI as your preferred C interface going forward.

It seems like we need to add ExpandEnvironmentStrings to `lib/chef/win32/api.rb` instead?